internal/pkg/client/gitlab/fetch_test.go

```diff
@@ -0,0 +1,201 @@
+package gitlab_test
+
+import (
+	"io"
+	"net/http"
+	"testing"
+
+	"github.com/stretchr/testify/mock"
+	gogitlab "github.com/xanzy/go-gitlab"
+)
+
+type mockClient struct {
+	mock.Mock
+}
+
+func (mc *mockClient) GetProject(
+	pid any,
+	opt *gogitlab.GetProjectOptions,
+	options ...gogitlab.RequestOptionFunc,
+) (*gogitlab.Project, *gogitlab.Response, error) {
+	args := mc.Called(pid, opt, options)
+
+	return args.Get(0).(*gogitlab.Project), args.Get(1).(*gogitlab.Response), args.Error(2) //nolint:revive,wrapcheck,lll
+}
+
+func (mc *mockClient) GetBranch(
+	pid any,
+	branch string,
+	options ...gogitlab.RequestOptionFunc,
+) (*gogitlab.Branch, *gogitlab.Response, error) {
+	args := mc.Called(pid, branch, options)
+
+	return args.Get(0).(*gogitlab.Branch), args.Get(1).(*gogitlab.Response), args.Error(2) //nolint:revive,wrapcheck,lll
+}
+
+func (mc *mockClient) GetCommit(
+	pid any,
+	sha string,
+	opt *gogitlab.GetCommitOptions,
+	options ...gogitlab.RequestOptionFunc,
+) (*gogitlab.Commit, *gogitlab.Response, error) {
+	args := mc.Called(pid, sha, opt, options)
+
+	return args.Get(0).(*gogitlab.Commit), args.Get(1).(*gogitlab.Response), args.Error(2) //nolint:revive,wrapcheck,lll
+}
+
+func (mc *mockClient) CreateDependencyListExport(
+	pipelineID int,
+	opt *gogitlab.CreateDependencyListExportOptions,
+	options ...gogitlab.RequestOptionFunc,
+) (*gogitlab.DependencyListExport, *gogitlab.Response, error) {
+	args := mc.Called(pipelineID, opt, options)
+
+	return args.Get(0).(*gogitlab.DependencyListExport), args.Get(1).(*gogitlab.Response), args.Error(2) //nolint:revive,wrapcheck,lll
+}
+
+func (mc *mockClient) GetDependencyListExport(
+	id int,
+	options ...gogitlab.RequestOptionFunc,
+) (*gogitlab.DependencyListExport, *gogitlab.Response, error) {
+	args := mc.Called(id, options)
+
+	return args.Get(0).(*gogitlab.DependencyListExport), args.Get(1).(*gogitlab.Response), args.Error(2) //nolint:revive,wrapcheck,lll
+}
+
+func (mc *mockClient) DownloadDependencyListExport(
+	id int,
+	options ...gogitlab.RequestOptionFunc,
+) (io.Reader, *gogitlab.Response, error) {
+	args := mc.Called(id, options)
+
+	return args.Get(0).(io.Reader), args.Get(1).(*gogitlab.Response), args.Error(2) //nolint:revive,wrapcheck
+}
+
+var successGitLabResponse = &gogitlab.Response{
+	Response: &http.Response{
+		Status:     "200 OK",
+		StatusCode: 200,
+		Proto:      "HTTP/1.0",
+	},
+}
+
+func TestCreateExport(t *testing.T) { //nolint:paralleltest
```

Comment:
These should be using a testify `suite.Suite` receiver. There's a VSCode snippet you can use to generate all the stubs for a suite, type `stretchr` and select the snippet suggestion to generate